### PR TITLE
asadiqbal08/Basic Product Detail Page

### PR DESCRIPTION
**Pull request by asadiqbal08**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
fixes: #32 

#### What's this PR do?
Design the basic product detail page.

#### Screenshots (if appropriate)

screencapture-localhost-8013-homepage-data-analysis-for-social-scientist-2021-08-04-17_26_56

Screen Shot 2021-08-03 at 4 52 40 PM

screencapture-localhost-8013-homepage-data-analysis-for-social-scientist-2021-08-04-17_27_40

## When price link is not there:

screencapture-localhost-8013-homepage-seed-digital-learning-course-2021-08-03-18_46_28

### CODE REVIEW

File: cms/migrations/0003_product_detail_page.py
Comment (pdpinch) [inline cms/migrations/0003_product_detail_page.py:44]: Can we make this a rich text field? It seems like courses on edX have bullet lists and links in their prerequisites. (I can't remember if rich text fields support links).

from https://www.edx.org/course/microeconomics:

image

**Review by pdpinch — CHANGES_REQUESTED** [review]
I think the price field is missing. Here it is in the (current) design:

image

Unlike xPRO, price should be a wagtail field where the CMS user can enter text or a link (as it appears in InVision).

File: cms/migrations/0003_product_detail_page.py
Comment (asadiqbal08) [reply cms/migrations/0003_product_detail_page.py:44]: I am going to add the price but I did not get what `Various` is pointing out and how to show the price if not having field in CMS  ?

Comment (asadiqbal08) [reply cms/migrations/0003_product_detail_page.py]: Field is richtext so can be used something like that. 


screencapture-localhost-8013-homepage-data-analysis-for-social-scientist-2021-08-02-15_43_46

File: cms/templates/product_page.html
Comment (pdpinch) [inline cms/templates/product_page.html:55]: Can we capitalize the text in CSS instead of in the HTML?

File: cms/templates/product_page.html
Comment (pdpinch) [inline cms/templates/product_page.html:57]: This text should be managed in the CMS

File: static/scss/product-details.scss
Comment (pdpinch) [inline static/scss/product-details.scss:9]: I thought we were moving to Montserrat? (not sure I spelled that right)

File: cms/templates/product_page.html
Comment (asadiqbal08) [reply cms/templates/product_page.html:57]: Thanks for answering this https://github.com/mitodl/mitxonline/pull/45#discussion_r680813560

File: static/scss/product-details.scss
Comment (asadiqbal08) [reply static/scss/product-details.scss:9]: Ok, I need to remove it from here , I guess, in one of the Arslan's PR it is added in base level.

**Comment by asadiqbal08** [comment]
@pdpinch I have updated the screenshots. Let me know if do you think to change anything ? 
Sorry, I have to squash the commits after rebased as many things need to push.

**Comment by pdpinch** [comment]
The price link should be optional. For MVP we will have a link, but after MVP some courses will have simple string for price like "$150." I think it's best to set it to optional now.

File: cms/templates/product_page.html
Comment (pdpinch) [inline cms/templates/product_page.html:61]:
```suggestion
              <span class="title">prerequisites</span>
```

should match all the design

**Comment by pdpinch** [comment]
Small thing, but in the design, the width of the "prerequisites" section is the same as the width of the video player. Is there a reasonable way to get them to line up? If it's too hard, we can leave it for a follow up.

**Review by pdpinch — CHANGES_REQUESTED** [review]

**Comment by asadiqbal08** [comment]
> Small thing, but in the design, the width of the "prerequisites" section is the same as the width of the video player. Is there a reasonable way to get them to line up? If it's too hard, we can leave it for a follow up.

> Out of scope:
    Video (a simple placeholder div can be used instead of a video in the template)


@pdpinch , In the screenshot, There is not any video but just an image holder. What do you think if we update it when start working on the video area there ?

**Comment by pdpinch** [comment]
>What do you think if we update it when start working on the video area there ?

sure

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:69]: We should try to avoid specialized styles for different header elements if we can. Does this really need a different font-size, line-height, or font-weight definition from the base styles that we have defined for `h1`? As far as I can tell, this is a header that should look like all other headers of the same size/type

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:76]: Adding to my comment above: if the designs call for the most prominent headers to be a certain size in mobile or desktop widths, those styles should be applied to the `h1` definition in `common.scss` unless this header is truly exceptional for some reason

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:109]: Not needed

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:215]: Please refer to my comment above about styling consistency with these header elements

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:163]: Can you wrap these `.text` and `.title` definition in a `.stats-col` selector? "text" and "title" are very general terms and the extra selector would help to understand what they're actually styling

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:184]: If we're using `display: block` and setting width to 100%, any reason we wouldn't just use a `div` instead of a `span`?

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:206]: Font family definition is not needed

**Review by gsidebo — CHANGES_REQUESTED** [review]
To add to the code comments: the `small` style applied to these elements is making the font sizes different between the titles. I don't think we want that

ss 2021-08-03 at 13 09 21

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:85]: One more comment: wherever possible we should be looking for styles that can be reused in other parts of the app. The "Enroll Now" button in the design has the same styling as many other buttons in other pages. You would expect any website to have some consistent styling for commonly-used elements like buttons, headers, borders, etc. This selector will style one single button in the product detail page, but looking at the designs, we know that this look and feel will be needed for other buttons. We should move these styles to a css class that any other button element can use (`.action-btn` or something like that)

**Review by gsidebo — CHANGES_REQUESTED** [review]

**Comment by asadiqbal08** [comment]
Updated the Screenshots

File: static/scss/product-details.scss
Comment (gsidebo) [inline static/scss/product-details.scss:6]:
```suggestion
  box-sizing: border-box;
```

1. I don't think the `padding` or `margin` rules are necessary here. I disabled both in mobile and desktop widths and it doesn't seem to have an effect
2. If this page needs to have a max-width that is greater than the standard width for bootstrap `container` elements, then that should be applied to _all_ top-level `container` elements. I think we should hold off on setting a new max-width for all elements and see what Ferdi, Peter, and the designer have to say. That can be done in a separate PR

**Review by gsidebo — APPROVED** [review]
I made one final [comment](https://github.com/mitodl/mitxonline/pull/45#discussion_r682738693) suggesting a simple css change. After you apply that, feel free to merge! 👍

**Pull request merged**
